tools/explorer/lib/ontology.py

```diff
@@ -31,35 +32,137 @@ def GetFieldsForTypeName(self,
                            namespace: str,
                            entity_type_name: str,
                            required_only: bool = False) -> List[ETF]:
-    """Gets a list of fields for a given typename within a namespace.
+    """Gets a list of fields for a given typename within a namespace
+
+       Args:
+            namespace: the name of the namespace as a string
+            entity_type_name: name of the entity type as a string
+            required_only: when true will return
+            only required fields for a given type
+
+       Returns:
+            result_fields: a list of StandardField tuples
+    """
+    entity_type = self.universe.entity_type_universe.GetEntityType(
+        namespace,
+        entity_type_name
+    )
+    qualified_fields = entity_type.GetAllFields()
+    field_optwrappers = [
+        entity_type.GetField(field)
+        for field in qualified_fields
+    ]
+    entity_type_fields = [
+        ETF(namespace,
+            field.field.field[0:],
+            not field.optional,
```

Comment:
generally speaking it's less confusing to keep the same sense for booleans throughout an application.  Is there any reason that we could use True == Optional here too?  (the logic behind making Optional == True is that we wanted things to default to being stricter - if you don't specify, it's required)

Reply:
Yeah this would make more sense. I'll make the change.